src/google/protobuf/compiler/objectivec/objectivec_file.cc

```diff
@@ -115,6 +115,9 @@ void FileGenerator::GenerateHeader(io::Printer *printer) {
   printer->Print(
       "// @@protoc_insertion_point(imports)\n"
       "\n"
+      "#pragma clang diagnostic push\n"
+      "#pragma clang diagnostic ignored \"-Wdeprecated-declarations\"\n"
+      "\n"
```

Comment:
Thanks, that's what I meant.

I was at first thinking it was only needed in the generated source, but if an enum is deprecated, the field that uses it would trigger it just from parsing the header.